pkg/controller/notification/notification_controller.go

```diff
@@ -166,12 +194,46 @@ func (r *ReconcileNotification) setStatusNotificationDeletionFailed(notification
 		})
 }
 
-func (r *ReconcileNotification) setStatusNotificationSent(notification *toolchainv1alpha1.Notification) error {
+func (r *ReconcileNotification) setStatusNotificationContextError(notification *toolchainv1alpha1.Notification, msg string) error {
 	return r.updateStatusConditions(
 		notification,
 		toolchainv1alpha1.Condition{
-			Type:   toolchainv1alpha1.NotificationSent,
-			Status: corev1.ConditionTrue,
-			Reason: toolchainv1alpha1.NotificationSentReason,
+			Type:    toolchainv1alpha1.NotificationSent,
+			Status:  corev1.ConditionFalse,
+			Reason:  toolchainv1alpha1.NotificationContextErrorReason,
+			Message: msg,
 		})
 }
+
+func (r *ReconcileNotification) setStatusNotificationDeliveryError(notification *toolchainv1alpha1.Notification, msg string) error {
+	return r.updateStatusConditions(
+		notification,
+		toolchainv1alpha1.Condition{
+			Type:    toolchainv1alpha1.NotificationSent,
+			Status:  corev1.ConditionFalse,
+			Reason:  toolchainv1alpha1.NotificationDeliveryErrorReason,
+			Message: msg,
+		})
+}
+
+func (r *ReconcileNotification) setStatusNotificationSent(notification *toolchainv1alpha1.Notification, msg string) error {
+	return r.updateStatusConditions(
+		notification,
+		toolchainv1alpha1.Condition{
+			Type:    toolchainv1alpha1.NotificationSent,
+			Status:  corev1.ConditionTrue,
+			Reason:  toolchainv1alpha1.NotificationSentReason,
+			Message: msg,
+		})
+}
+
+func (r *ReconcileNotification) updateStatus(logger logr.Logger, notification *toolchainv1alpha1.Notification,
+	statusUpdater func(notification *toolchainv1alpha1.Notification, msg string) error) error {
```

Comment:
same here using the `statusUpdater` type suggested above

Reply:
changed here: 057eb87e2123a2edfa90a49c007140b845d59010